src/shared/ui/survey/CheckBox/CheckBoxActivity.item.tsx

```diff
@@ -0,0 +1,70 @@
+import { FC, useMemo } from 'react';
+
+import { shuffle } from '@shared/lib';
+import { Box, ScrollView } from '@shared/ui';
+
+import CheckBoxItem from './CheckBox.item';
+import { Item } from './types';
+
+type CheckBoxProps = {
+  // @todo make sure backend will update config to new keys described below (type RefactoredConfig in ./types)
+  config: {
+    itemList: Item[];
+    minValue: number;
+    maxValue: number;
+    colorPalette: boolean;
+    randomizeOptions: boolean;
+  };
+  onChange: (arrayOfValues: number[]) => void;
+  values: number[];
+};
+
+const CheckBoxActivityItem: FC<CheckBoxProps> = ({
+  config,
+  onChange,
+  values = [],
+}) => {
+  const { minValue, maxValue } = config;
+  const hasSingleItem = maxValue === 1 && minValue === 1;
+
+  const { itemList, randomizeOptions } = config;
+
+  const items = useMemo(() => {
+    const filteredItems = itemList.filter(({ isVis }) => !isVis);
```

Comment:
We do we use isVis name in the refactored version? 
We are able to use normal names like isVisible